libpod/container.go

```diff
@@ -407,6 +409,58 @@ func (c *Container) teardownStorage() error {
 	return nil
 }
 
+func (c *Container) addImageData(g *generate.Generator) (err error) {
+	var (
+		args []string
+	)
+	if !c.config.UseImageConfig {
+		return nil
+	}
+	imageData, err := c.runtime.GetImageConfig(c.config.RootfsImageID)
+	if err != nil {
+		return errors.Wrapf(err, "error reading container image data")
+	}
+	if imageData.User != "" {
+		uid, _, err := chrootuser.LookupUserInContainer(c.bundlePath(), imageData.User)
+		if err != nil {
+			return errors.Wrapf(err, "user %q is not defined in %q", imageData.User, c.bundlePath())
+		}
+		g.SetProcessUID(uint32(uid))
+	}
+	for _, e := range imageData.ExposedPorts {
+		//TODO exposed ports are not handled yet
+		logrus.Warn("Exposed port %q is ignored", e)
+	}
+
+	if len(imageData.Cmd) > 0 {
+		args = imageData.Cmd
+	}
+	if len(imageData.Entrypoint) > 0 {
+		args = imageData.Entrypoint
+	}
+	if len(args) > 0 {
+		g.SetProcessArgs(args)
+	}
+	if imageData.StopSignal != "" {
+		g.AddAnnotation("org.opencontainers.image.stopSignal", imageData.StopSignal)
+	}
+	for _, v := range imageData.Volumes {
+		//TODO volumes are not handled yet
+		logrus.Warn("Builtin Volume %q is ignored", v)
+	}
+	if imageData.WorkingDir != "" {
+		g.SetProcessCwd(imageData.WorkingDir)
+	}
+	for _, env := range imageData.Env {
```

Comment:
As with ports, I don't know how we handle this if the user already set environment data. Do we add the environment from the image, or just drop the image's environment vars? Need to test with docker, but I feel like we should add them only if they aren't already present in the spec.

Reply:
These are additive.  If the user sets the same then the user wins.  But if the container image sets ABC=def, and the user sets XYZ=nqp then both gets set.